src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -774,4 +728,81 @@ private function getSerializerContext(string $operationType, bool $denormalizati
 
         return $resourceMetadata->getItemOperationAttribute($operationName, $contextKey, null, true);
     }
+
+    /**
+     * @param ResourceMetadata $resourceMetadata
+     * @param string           $operationName
+     * @param \ArrayObject     $pathOperation
+     */
+    private function addPaginationParameters(ResourceMetadata $resourceMetadata, string $operationName, \ArrayObject $pathOperation)
+    {
+        if ($this->paginationEnabled && $resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_enabled', true, true)) {
+            $pathOperation['parameters'][] = $this->getPaginationParameters();
+
+            if ($resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_client_items_per_page', $this->clientItemsPerPage, true)) {
+                $pathOperation['parameters'][] = $this->getItemsParPageParameters();
+            }
+        }
+    }
+
+    /**
+     * @param array            $subresourceOperation
+     * @param \ArrayObject     $definitions
+     * @param string           $operationId
+     * @param array            $mimeTypes
+     * @param string           $resourceClass
+     * @param ResourceMetadata $resourceMetadata
+     *
+     * @throws \ApiPlatform\Core\Exception\ResourceClassNotFoundException
+     *
+     * @return \ArrayObject
+     */
+    private function addSubresourceOperation(array $subresourceOperation, \ArrayObject $definitions, string $operationId, array $mimeTypes, string $resourceClass, ResourceMetadata $resourceMetadata): \ArrayObject
+    {
+        $operationName = 'get';
+        $subResourceMetadata = $this->resourceMetadataFactory->create($subresourceOperation['resource_class']);
+        $serializerContext = $this->getSerializerContext(OperationType::SUBRESOURCE, false, $subResourceMetadata, $operationName);
+        $responseDefinitionKey = $this->getDefinition($definitions, $subResourceMetadata, $subresourceOperation['resource_class'], $serializerContext);
+
+        $pathOperation = new \ArrayObject([]);
+        $pathOperation['tags'] = $subresourceOperation['shortNames'];
+        $pathOperation['operationId'] = $operationId;
+        $pathOperation['produces'] = $mimeTypes;
+        $pathOperation['summary'] = sprintf('Retrieves %s%s resource%s.', $subresourceOperation['collection'] ? 'the collection of ' : 'a ', $subresourceOperation['shortNames'][0], $subresourceOperation['collection'] ? 's' : '');
+        $pathOperation['responses'] = [
```

Comment:
We usually avoid to use ternaries if the expression doesn't fit on one line, but has it was already like that previously, fell free to keep this as is if you want.